From 5aae5e0ec33c99b317fd501080e92e52292313b6 Mon Sep 17 00:00:00 2001
From: Debasish Biswas <debasishbsws.dev@gmail.com>
Date: Wed, 7 May 2025 22:38:31 +0530
Subject: [PATCH] Fixes: yaml-cpp target not found during build

The RStudio source code expected to link against the modern namespaced CMake target yaml-cpp::yaml-cpp, but our system provides the traditional target name yaml-cpp. This patch modifies the CMakeLists.txt to use the non-namespaced target name so that it correctly links against the system-provided yaml-cpp library.

Signed-off-by: Debasish Biswas <debasishbsws.dev@gmail.com>
---
 src/cpp/core/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cpp/core/CMakeLists.txt b/src/cpp/core/CMakeLists.txt
index 563f4c382e..eb0fa75760 100644
--- a/src/cpp/core/CMakeLists.txt
+++ b/src/cpp/core/CMakeLists.txt
@@ -336,7 +336,7 @@ target_link_libraries(rstudio-core
    fmt::fmt
    gsl::gsl-lite-v1
    tl::expected
-   yaml-cpp::yaml-cpp
+   yaml-cpp
 )
 
 if(WIN32)
-- 
2.39.5 (Apple Git-154)

